addons/stock/models/stock_quant.py

```diff
@@ -1393,7 +1416,8 @@ def unpack(self):
             ])
             move_line_to_modify.write({'package_id': False})
             unpacked_quants |= package.quant_ids
-            package.mapped('quant_ids').sudo().write({'package_id': False})
+            package.quant_ids._move_quants(message=_("Quantities unpacked"), out=True)
```

Comment:
you can remove upper lines since unreservation will be done in action_done of sml